### ISSUE TITLE
FR: `jj op undo $n`

### ISSUE THREAD

**Issue by Zoybean** [jj-vcs/jj#4056]
**Is your feature request related to a problem? Please describe.**
Most often when I'm undoing something, I'll look at the op log and then figure out the number of operations I want to undo, then count that many `-`'s, e.g.: `jj op restore @---`.
I rarely have a use for more than 2 `+` or `-`, outside of `jj op restore`.

**Describe the solution you'd like**
I'd like to be able to say `jj op undo 3` to undo the last 3 operations

**Describe alternatives you've considered**
`jj op restore @-3` using the syntax in #4055 and #3337

**Comment by yuja** [comment]
See also https://github.com/martinvonz/jj/issues/3700 for broader discussion.

**Comment by PhilipMetzger** [comment]
Yes, this should be folded into the new high level `jj undo` (#3700), as `jj op undo` is to low level for such a convenience detail

**Comment by PhilipMetzger** [comment]
Now that the higher level undo exists, this should be a good first contribution. Since all you need to do is to add a `depth` parameter both to `undo/redo` and plumb it through the logic.

**Comment by senekor** [comment]
I personally think this should not be added. Now that `jj undo` is more high-level, I don't see the need for this. It's hard to know in advance how many operations you need to undo, because there could be snapshot operations that the user doesn't mentally keep track of all the time. So you kinda _need_ to read `jj op log` before running something like `jj undo $n`. So that breaks the high-level ergonomics of `jj undo` in my opinion.

If a user needs to undo so many operations that hitting "up+enter" a bunch of times is too tedious, I think they should copy the operation ID and `jj op restore $ID` directly?

So, currently I'm in favor of closing as "not planned".
